Docs/prewritten policies (#792)
hashicorp/terraform-docs-common: website/docs/cloud-docs/policy-enforcement/import-reference/index.mdx

---
page_title: Sentinel import function reference
description: Use the Sentinel import function to configure policy behaviors in custom Sentinel policies. 
---
# `import` function reference overview

> This topic provides an overview of the Sentinel `import` functions you can use in custom Sentinel policies. Refer to [Define custom Sentinel policies](/terraform/cloud-docs/policy-enforcement/define-policies/custom-sentinel) for additional information about how to use the functions.

Reviewer: ```suggestion
This topic provides an overview of the Sentinel `import` function, which you can use to import Sentinel libraries into your custom Sentinel policies. Refer to [Define custom Sentinel policies](/terraform/cloud-docs/policy-enforcement/define-policies/custom-sentinel) for additional information about how to use the import function.
```

## Functions for Terraform

> You can include Sentinel `import` functions that enable a policy to access reusable libraries, external data, and functions. Refer to the [Sentinel imports documentation](/sentinel/docs/language/imports) for more details.

Reviewer: ```suggestion
You can include Sentinel `import` function that enable a policy to access reusable libraries, external data, and functions. Refer to the [Sentinel imports documentation](/sentinel/docs/language/imports) for more details.
```

I think it's just one "import" function used in different ways?

> HCP Terraform provides the following `import` functions to define policy rules for the plan, configuration, state, and run associated with a policy check.
> 
> - [`tfplan`](/terraform/cloud-docs/policy-enforcement/import-reference/tfplan) : Provides access to a Terraform plan, which is the file created when you run the `terraform plan` command. This function is deprecated. Use `tfplanv/2` instead.

Reviewer: ```suggestion
HCP Terraform provides the following importable libraries to define policy rules for the plan, configuration, state, and run associated with a policy check.

- [`tfplan`](/terraform/cloud-docs/policy-enforcement/import-reference/tfplan) : Provides access to a Terraform plan, which is the file created when you run the `terraform plan` command. This library is deprecated. Use `tfplanv/2` instead.
```
Again, I thought this was the correct terminology, ignore me if I'm wrong! :)

Otherwise, same comment applies to the rest of the list.

Author: No, thank you for addressing this. I wasn't sure what the correct terminology was. I just know that we shouldn't refer to them as "imports", like we're bringing cheese over from Italy.

> - [`tfplan/v2`](/terraform/cloud-docs/policy-enforcement/import-reference/tfplan-v2): Provides access to a Terraform plan, which is the file created when you run the `terraform plan` command. 
> - [`tfconfig`](/terraform/cloud-docs/policy-enforcement/import-reference/tfconfig): Provides access to a Terraform configuration. The configuration is the set of `.tf` files that describe the desired infrastructure state. This function is deprecated. Use `tfconfig/v2` instead.
> - [`tfconfig/v2`](/terraform/cloud-docs/policy-enforcement/import-reference/tfconfig-v2): Provides access to a Terraform configuration. The configuration is the set of `.tf` files that describe the desired infrastructure state. This function is deprecated. Use `tfconfig/v2` instead.
> - [`tfstate`](/terraform/cloud-docs/policy-enforcement/import-reference/tfstate): Provides access to the Terraform state. Terraform uses state to map real-world resources to your configuration. This function is deprecated. Use `tfstate/v2` instead.
> - [`tfstate/v2`](/terraform/cloud-docs/policy-enforcement/import-reference/tfstate-v2): Provides access to the Terraform state. Terraform uses state to map real-world resources to your configuration. 
> - [`tfrun`](/terraform/cloud-docs/policy-enforcement/import-reference/tfrun): Provides access to data associated with a run in HCP Terraform. For example, you could retrieve the run's workspace.

Author: ```suggestion
- [`tfplan/v2`](/terraform/cloud-docs/policy-enforcement/import-reference/tfplan-v2): Provides access to a Terraform plan, which is the file created when you run the `terraform plan` command. 
- [`tfconfig`](/terraform/cloud-docs/policy-enforcement/import-reference/tfconfig): Provides access to a Terraform configuration. The configuration is the set of `.tf` files that describe the desired infrastructure state. This library is deprecated. Use `tfconfig/v2` instead.
- [`tfconfig/v2`](/terraform/cloud-docs/policy-enforcement/import-reference/tfconfig-v2): Provides access to a Terraform configuration. The configuration is the set of `.tf` files that describe the desired infrastructure state. This library is deprecated. Use `tfconfig/v2` instead.
- [`tfstate`](/terraform/cloud-docs/policy-enforcement/import-reference/tfstate): Provides access to the Terraform state. Terraform uses state to map real-world resources to your configuration. This library is deprecated. Use `tfstate/v2` instead.
- [`tfstate/v2`](/terraform/cloud-docs/policy-enforcement/import-reference/tfstate-v2): Provides access to the Terraform state. Terraform uses state to map real-world resources to your configuration. 
- [`tfrun`](/terraform/cloud-docs/policy-enforcement/import-reference/tfrun): Provides access to data associated with a run in HCP Terraform. For example, you could retrieve the run's workspace.
```

> ## Test `import` functions
> You can create mocks of these functions and test them using the Sentinel CLI. Refer to the following topics for additional information:
> [Test Sentinel policies](/terraform/cloud-docs/policy-enforcement/test-sentinel)
> [Sentinel CLI documentation](/sentinel/docs/commands)

Reviewer: ```suggestion
## Test `import` functions
You can create mocks of these functions and test them using the Sentinel CLI. Refer to the following topics for additional information:

- [Test Sentinel policies](/terraform/cloud-docs/policy-enforcement/test-sentinel)
- [Sentinel CLI documentation](/sentinel/docs/commands)
```
